addon/components/sl-button.js

```diff
@@ -145,45 +156,57 @@ export default Ember.Component.extend( TooltipEnabled, {
         if ( label ) {
             return label;
         }
-    }.property( 'label', 'pending', 'pendingLabel' ),
+    }),
 
     /**
      * Converted size string to Bootstrap button class
      *
-     * @function sizeClass
+     * @function
      * @observes size
-     * @returns  {Ember.String} Defaults to undefined
+     * @returns {?String} Defaults to undefined
      */
-    sizeClass: function() {
+    sizeClass: Ember.computed( 'size', function() {
         var size = this.get( 'size' ),
             sizeClass;
 
+        Ember.assert(
+            'Error: Invalid size value',
+            Object.keys( SIZE ).map( ( key ) => SIZE[ key ] ).indexOf( size ) > -1
+        );
+
         switch ( size ) {
-            case 'extra-small':
+            case SIZE.EXTRA_SMALL:
                 sizeClass = 'btn-xs';
                 break;
 
-            case 'small':
+            case SIZE.SMALL:
                 sizeClass = 'btn-sm';
                 break;
 
-            case 'large':
+            case SIZE.LARGE:
                 sizeClass = 'btn-lg';
                 break;
         }
 
         return sizeClass;
-    }.property( 'size' ),
+    }),
 
     /**
      * Converted theme string to Bootstrap button class
      *
-     * @function themeClass
+     * @function
      * @observes theme
```

Comment:
@observes can be removed due to the intended use of https://github.com/notmessenger/jsdoc-plugins#emberobservestag